shell/platform/windows/platform_handler.cc

```diff
@@ -340,11 +353,61 @@ void PlatformHandler::SystemSoundPlay(
   }
 }
 
+void PlatformHandler::SystemExitApplication(
+    const std::string& exit_type,
+    int64_t exit_code,
+    std::unique_ptr<MethodResult<rapidjson::Document>> result) {
+  rapidjson::Document result_doc;
+  result_doc.SetObject();
+  if (exit_type.compare(kExitTypeRequired) == 0) {
+    QuitApplication(exit_code);
+    result_doc.GetObjectW().AddMember(kExitResponseKey, kExitResponseExit,
+                                      result_doc.GetAllocator());
+    result->Success(result_doc);
+  } else {
+    RequestAppExit(exit_type, exit_code);
+    result_doc.GetObjectW().AddMember(kExitResponseKey, kExitResponseCancel,
+                                      result_doc.GetAllocator());
+    result->Success(result_doc);
+  }
+}
+
+void PlatformHandler::RequestAppExit(const std::string& exit_type,
+                                     int64_t exit_code) {
+  auto callback = std::make_unique<MethodResultFunctions<rapidjson::Document>>(
+      [this, exit_code](const rapidjson::Document* response) {
+        RequestAppExitSuccess(response, exit_code);
```

Comment:
I'm confused by the protocol here. The documentation for `RequestAppExit` says:

> Send a request to the framework to test if a cancelable exit request should be canceled or honored.

That suggests to me that there are two possible responses: exit, or don't exit. But this unconditionally calls `RequestAppExitSuccess` on any response, which unconditionally exits, and whose documentation is:

> Callback from when the cancelable exit request response request is answered by the framework.

How would the Dart side respond to cancel the exit? It sounds suspiciously like the way to not exit is to not respond, but it's a programming error (leak) to never respond to a method call.

Reply:
`RequestAppExitSuccess` does not unconditionally exit. It exits iff `(exit_type.compare(kExitResponseExit) == 0)`. The dart side responds with the `response` argument set to either `exit` or `cancel`, the latter of which does not exit the application.

Reply:
🤦🏻 Sorry, I'm not sure how I missed that.